File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/payment/PaymentGateway.java
package com.fixture.shop.payment;

/**
 * Payment processing contract. goToImplementation should resolve to
 * StripeGateway, PayPalGateway, MockGateway.
 */
public interface PaymentGateway {

    PaymentResult charge(double amount, String currency);

    PaymentResult refund(String transactionId);
}

File: skill_sdk/tests/fixtures/ts-project/src/shop/payment/PaymentGateway.ts
import { PaymentResult } from './PaymentResult';

/**
 * Payment processing contract. goToImplementation should resolve to
 * StripeGateway, PayPalGateway, MockGateway.
 */
export interface PaymentGateway {
  charge(amount: number, currency: string): PaymentResult;
  refund(transactionId: string): PaymentResult;
}
